pkg/driver/validation.go

```diff
@@ -76,6 +76,7 @@ func validateExtraTags(tags map[string]string, warnOnly bool) error {
 		if strings.HasPrefix(k, cloud.AWSTagKeyPrefix) {
 			return fmt.Errorf("Tag key prefix '%s' is reserved", cloud.AWSTagKeyPrefix)
 		}
+		//TODO possibly remove these lines below as failure condition is impossible to encounter
```

Comment:
Start of discussion thread

Reply:
Why are they impossible to encounter? Wouldn't something like a `$` (or any other non-allowed character) in the key/value hit the regex? Or is that checked elsewhere?

Reply:
Following up on our in person discussion as in line with [this doc ](https://docs.aws.amazon.com/mediaconnect/latest/ug/tagging-restrictions.html#:~:text=Allowed%20characters%20are%20letters%2C%20numbers%2C%20spaces%20representable%20in%20UTF%2D8%2C%20and%20the%20following%20characters%3A%20.%20%3A%20%2B%20%3D%20%40%20_%20/%20%2D%20(hyphen).%20Amazon%20EC2%20resources%20allow%20any%20characters.) EC2 Resources can have any utf-8 valid tag although currently the regex we are using is bugged as it includes the empty string making any key or value checked count as a match no matter what. I have confirmed that ebs does fall under this category of EC2 resources by creating a volume myself and tagging it with the tag $$$ 


![Screenshot 2024-08-05 at 3 43 33 PM](https://github.com/user-attachments/assets/6a0afe5c-1346-4c36-b8a0-582d1fabb4cc)


As such I will make a commit to remove the regex and the lines testing for the regex.

Reply:
The doc actual mentions ec2 resources can use **any** character not just any utf-8 which I had assumed when reading it at first to further test I found a list of non utf-8 characters and they did in fact work as a tag not sure if there is any niche edge cases that we are missing though .
![Screenshot 2024-08-06 at 9 58 03 AM](https://github.com/user-attachments/assets/cde20159-a4d4-475e-ad2a-2028870d4a73)
****